mathics/builtin/intfns/combinatorial.py

```diff
@@ -39,6 +40,35 @@
 SymbolSubsets = Symbol("Subsets")
 
 
+class BellB(SympyFunction):
+    """
+    <url>:Bell number: https://en.wikipedia.org/wiki/Bell_number</url>
+    (<url>:SymPy: https://docs.sympy.org/latest/modules/functions/combinatorial.html#sympy.functions.combinatorial.numbers.bell</url>,
+    <url>:WMA: https://reference.wolfram.com/language/ref/BellB.html</url>)
+    <dl>
+      <dt>'BellB[$n$]'
+      <dd>Bell number $B_n$.
+
+      <dt>'BellB[$n$, $x$]'
+      <dd>Bell polynomial $B_n(x)$.
+    </dl>
+
+    >> BellB[10]
+     = 115975
+
+    >> BellB[5, x]
+     = x + 15 x ^ 2 + 25 x ^ 3 + 10 x ^ 4 + x ^ 5
+    """
+
+    attributes = A_LISTABLE | A_NUMERIC_FUNCTION | A_PROTECTED
```

Comment:
In WMA, `Attributes[BellB]={Listable, NHoldFirst, NumericFunction, Protected, ReadProtected}`

Reply:
(`ReadProtected` is not needed, but `NHoldFirst` affects the evaluation)